src/layers.c

```diff
@@ -118,6 +86,34 @@ handle_output_destroy(struct wl_listener *listener, void *data)
 	wlr_layer_surface_v1_destroy(layer->scene_layer_surface->layer_surface);
 }
 
+static void
+process_keyboard_interactivity(struct lab_layer_surface *layer)
+{
+	struct wlr_layer_surface_v1 *layer_surface = layer->scene_layer_surface->layer_surface;
+	struct seat *seat = &layer->server->seat;
+
+	if (layer_surface->current.keyboard_interactive
+			&& layer_surface->current.layer >= ZWLR_LAYER_SHELL_V1_LAYER_TOP) {
+		/*
+		 * Give keyboard focus to surface if
+		 * - keyboard-interactivity is 'exclusive' or 'on-demand'; and
+		 * - surface is in top/overlay layers; and
+		 * - currently focused layer has a lower precedence
+		 */
+		if (!seat->focused_layer || seat->focused_layer->current.layer
+				>= layer_surface->current.layer) {
```

Comment:
This seems to contradict the comment above, shouldn't it be `seat->focused_layer->current.layer < layer_surface->current.layer` to match `currently focused layer has a lower precedence`?

Reply:
You're 100% right. Thanks you. The power of peer review!
I've changed this + extended the comment to make it super clear.